scripts/docs-collator/ModuleFetcher.py

```diff
@@ -40,9 +40,18 @@ def __fetch_images(self, repo, module_download_dir):
 
     def __fetch_submodules(self, repo, module_download_dir):
         remote_files = self.github_provider.list_repo_dir(repo, SUBMODULES_DIR)
+        readme_files = {}
 
         for remote_file in remote_files:
-            if os.path.basename(remote_file) != README_MD:
-                continue
-
-            self.github_provider.fetch_file(repo, remote_file, module_download_dir)
+            base_name = os.path.basename(remote_file)
+            dir_name = os.path.dirname(remote_file)
+
+            if base_name == README_YAML:
+                readme_files[dir_name] = remote_file
+            elif base_name == README_MD and dir_name not in readme_files:
+                readme_files[dir_name] = remote_file
```

Comment:
```suggestion
            if base_name == README_YAML or base_name == README_MD and dir_name not in readme_files:
                readme_files[dir_name] = remote_file
```
Since both blocks do the same thing, it's best to condense this

Reply:
@dudymas What is the final value of `readme_files[dirname]` in both the original code and the suggested code when the loop provides these values for `base_name` in this order:
- `README.yaml`
- `README.md`

Reply:
the final value would likely be `README.yaml` for either code. I'd assume that suggested code would not change behavior since both sides of the `or` would evaluate to false (the `dir_name` would be in `readme_files` since both files have the same `dir_name`.)

Reply:
Regardless, I'm not here to suggest running tests. If the code is difficult to test, the refactoring isn't worth it.

Reply:
@dudymas Yes, it would be `README.yaml` in either code. But if the list were in alphabetical order:

- `README.md`
- `README.yaml`

Then the result of the original code would still be `README.yaml`, while the suggested replacement would yield `README.md`.